components/motor/roboclaw/roboclaw.go

```diff
@@ -239,14 +239,13 @@ func (m *roboclawMotor) SetPower(ctx context.Context, powerPct float64, extra ma
 }
 
 func goForMath(rpm, revolutions float64) (float64, time.Duration) {
-	// If revolutions is 0, the returned wait duration will be 0 representing that
-	// the motor should run indefinitely.
-	if revolutions == 0 {
-		powerPct := rpm / maxRPM
-		return powerPct, 0
+	dir := 1.0
+	if rpm*revolutions == 0.0 {
+		dir = 0.0
```

Comment:
similar recommendation, we can return early here, especially because the logic below breaks when `rpm == 0`